pkg/controller/logstorage/linseed.go

```diff
@@ -73,12 +74,25 @@ func (r *ReconcileLogStorage) createLinseed(
 	}
 	trustedBundle := certificateManager.CreateTrustedBundle(esInternalCertificate, prometheusCertificate)
 
+	// This secret should only ever contain one key.
+	if len(esAdminUserSecret.Data) != 1 {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, "Elasticsearch admin user secret contains too many entries", nil, reqLogger)
```

Comment:
In theory, data could be of length 0, in which case the error should say "not enough entries"

Maybe just adjust the log to say "secret should have exactly one entry. found %d"